### PR TITLE
Merge to main workflow

### PR DESCRIPTION
**Pull request by luckslovez**
This PR adds a workflow when merging to `main`;
- Publish linux (staging & prod)
  - PR workflow add the schema file required to publish artifacts to CAOS repos
  - merge_to_main workflow uses that with `newrelic/infrastructure-publish-action@v1.2.3` to publish linux pkgs
- Publish windows
  - The matrix generates windows-only matrix
  - M2M workflow uses the matrix to fetch windows artifacts form the pre-release to then upload them to logging's s3 bucket

### CODE REVIEW

File: .github/workflows/merge_to_main.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+name: Merge to main
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  get-release-tag:
+    name: Publish linux artifacts into s3 bucket
```
Comment (jsubirat) [inline .github/workflows/merge_to_main.yml:10]:
```suggestion
    name: Gets pre-release tag of the recently merged PR
```

File: .github/workflows/merge_to_main.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        with:
+          script: |
+            return (
+              await github.rest.repos.listPullRequestsAssociatedWithCommit({
+                commit_sha: context.sha,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+              })
+            ).data[0].number;
+          result-encoding: string
```
Comment (jsubirat) [inline .github/workflows/merge_to_main.yml:27]: Don't we get this payload as part of the GH workflow request (`context`) directly? 🤔

File: .github/workflows/merge_to_main.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          AWS_ROLE_ARN: ${{ secrets.OHAI_AWS_ROLE_ARN_STAGING }}
+          AWS_ROLE_SESSION_NAME: ${{ secrets.OHAI_AWS_ROLE_SESSION_NAME_STAGING }}
+        with:
+          tag: ${{env.TAG}}
+          repo_name: "newrelic/fluent-bit-package"
+          schema: "custom"
+          schema_url: "https://github.com/newrelic/fluent-bit-package/releases/download/${{ env.TAG }}/generated-linux-schema.yaml"
+          aws_access_key_id: ${{ env.AWS_ACCESS_KEY_ID }}
+          aws_secret_access_key: ${{ env.AWS_SECRET_ACCESS_KEY }}
+          aws_s3_bucket_name: ${{ env.AWS_S3_BUCKET_NAME }}
```
Comment (jsubirat) [inline .github/workflows/merge_to_main.yml:55]: Why is this (and many other properties) duplicated in the `env` and in the `with` blocks?

File: .github/workflows/merge_to_main.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+          AWS_ACCESS_KEY_ID: ${{ secrets.OHAI_AWS_ACCESS_KEY_ID_PRODUCTION }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.OHAI_AWS_SECRET_ACCESS_KEY_PRODUCTION }}
+          AWS_ROLE_ARN: ${{ secrets.OHAI_AWS_ROLE_ARN_PRODUCTION }}
+          AWS_ROLE_SESSION_NAME: ${{ secrets.OHAI_AWS_ROLE_SESSION_NAME_PRODUCTION }}
+        with:
+          tag: ${{env.TAG}}
+          repo_name: "newrelic/fluent-bit-package"
+          schema: "custom"
+          schema_url: "https://github.com/newrelic/fluent-bit-package/releases/download/${{ env.TAG }}/generated-linux-schema.yaml"
+          aws_access_key_id: ${{ env.AWS_ACCESS_KEY_ID }}
```
Comment (jsubirat) [inline .github/workflows/merge_to_main.yml:85]: Same question here 🤔

File: versions/strategyMatrix.py [hunk 5]
```diff
@@ -70,6 +74,8 @@ def deb_package_details(pkg):
         'targetPackageName': target_package_name,
         'nrPackageUrl':
 f"https://nr-downloads-main.s3.amazonaws.com/infrastructure_agent/linux/apt/pool/main/f/fluent-bit/{target_package_name}",
+        'repoArch': f"{pkg['arch']}",
```
Comment (jsubirat) [inline versions/strategyMatrix.py:77]: Why not simply...
```suggestion
        'repoArch': pkg['arch'],
```
?

File: versions/strategyMatrix.py [hunk 6]
```diff
@@ -91,6 +99,8 @@ def sles_package_details(pkg):
         'targetPackageName': target_package_name,
         'nrPackageUrl':
             f"https://nr-downloads-main.s3.amazonaws.com/infrastructure_agent/linux/zypp/{pkg['osDistro']}/{pkg['osVersion']}/{pkg['arch']}/{target_package_name}",
+        'repoArch': f"{pkg['arch']}",
```
Comment (jsubirat) [inline versions/strategyMatrix.py:102]: Same comment

**Review by jsubirat — COMMENTED** [review]
Some minor comments. Amazing work @luckslovez !

File: versions/strategyMatrix.py (hunk 5 above)
Comment (luckslovez) [reply versions/strategyMatrix.py:77]: Without the formatting is not treated as string apparently

File: .github/workflows/merge_to_main.yml (hunk 3 above)
Comment (luckslovez) [reply .github/workflows/merge_to_main.yml:55]: This is how we used to do it in other workflows

Comment (jsubirat) [reply .github/workflows/merge_to_main.yml:55]: Can we probably simplify this and avoid duplication (either `with` or `env`)?

**Review by jsubirat — APPROVED** [review]

**Pull request merged**